### PR TITLE
added support for 96 x 96 ssd1351 mikroe board

### PR DESCRIPTION
**Pull request by leragequit**
added support for 96 x 96 ssd1351 mikroe board
framebuffer was mirrored and viewport was misaligned. 
all examples work on the mikroe OLED C click using this driver.

### CODE REVIEW

**Comment by thijstriemstra** [comment]
@leragequit thanks! please check anf fix why travis is failing with this PR: https://travis-ci.org/rm-hull/luma.oled/jobs/329521634

File: luma/oled/device.py [hunk 1]
```diff
@@ -333,19 +333,24 @@ def __init__(self, serial_interface=None, width=128, height=128, rotate=0,
         self.capabilities(width, height, rotate, mode="RGB")
         self.framebuffer = getattr(luma.core.framebuffer, framebuffer)(self)
 
-        if width != 128 or height != 128:
+        settings = {
+            (128,128): dict(width=0x7f, height=0x7f, displayoffset=0x00, startline=0x00, remap=0x00),
+            (96,96): dict(width=0x6f, height=0x5f, displayoffset=0x00, startline=0x00, remap=0x02)
```
Comment (rm-hull) [inline luma/oled/device.py]: For consistency, use uppercase hex numbers

File: luma/oled/device.py [hunk 2]
```diff
@@ -333,19 +333,24 @@ def __init__(self, serial_interface=None, width=128, height=128, rotate=0,
...
         self.command(0xFD, 0x12)              # Unlock IC MCU interface
         self.command(0xFD, 0xB1)              # Command A2,B1,B3,BB,BE,C1 accessible if in unlock state
         self.command(0xAE)                    # Display off
         self.command(0xB3, 0xF1)              # Clock divider
-        self.command(0xCA, 0x7F)              # Mux ratio
-        self.command(0x15, 0x00, 0x7F)        # Set column address
-        self.command(0x75, 0x00, 0x7F)        # Set row address
-        self.command(0xA0, 0x74)              # Segment remapping
-        self.command(0xA1, 0x00)              # Set Display start line
+        self.command(0xCA, 0x7f)              # Mux ratio
```
Comment (rm-hull) [inline luma/oled/device.py]: For consistency, use uppercase hex numbers

File: luma/oled/device.py [hunk 3]
```diff
@@ -333,19 +333,24 @@ def __init__(self, serial_interface=None, width=128, height=128, rotate=0,
...
+        self.command(0x15, settings['displayoffset'], settings['width'])        # Set column address
+        self.command(0x75, settings['displayoffset'], settings['height'])        # Set row address
+        self.command(0xA0, 0x74 | settings['remap'])              # Segment remapping # Column address remapping or else everthing is mirrored
```
Comment (rm-hull) [inline luma/oled/device.py]: try and vertically align colums

File: luma/oled/device.py [hunk 4]
```diff
@@ -374,6 +379,10 @@ def display(self, image):
 
         if self.framebuffer.redraw_required(image):
             left, top, right, bottom = self.framebuffer.bounding_box
+            _,_,w,h = self.bounding_box
```
Comment (rm-hull) [inline luma/oled/device.py]: here are `h-offset` and `v-offset` params used elsewhere which we should emulate here. See: https://github.com/rm-hull/luma.lcd/commit/116c09abec7e3ab388699b2869ce9f07fd32a50a

File: luma/oled/device.py [hunk 5]
```diff
@@ -333,28 +333,33 @@ def __init__(self, serial_interface=None, width=128, height=128, rotate=0,
...
-        self.command(0xA2, 0x00)              # Set display offset
-        self.command(0xB5, 0x00)              # Set GPIO
-        self.command(0xAB, 0x01)              # Function select (internal - diode drop)
-        self.command(0xB1, 0x32)              # Precharge
-        self.command(0xB4, 0xA0, 0xB5, 0x55)  # Set segment low voltage
-        self.command(0xBE, 0x05)              # Set VcomH voltage
-        self.command(0xC7, 0x0F)              # Contrast master
-        self.command(0xB6, 0x01)              # Precharge2
-        self.command(0xA6)                    # Normal display
+        self.command(0xFD, 0x12)                                                    # Unlock IC MCU interface
```
Comment (thijstriemstra) [inline luma/oled/device.py]: can we move these comments and place them above the lines?

**Comment by thijstriemstra** [comment]
@leragequit travis is reporting these style issues:

```
./luma/oled/device.py:337:17: E231 missing whitespace after ','
./luma/oled/device.py:338:16: E231 missing whitespace after ','
./luma/oled/device.py:341:21: E711 comparison to None should be 'if cond is None:'
./luma/oled/device.py:382:14: E231 missing whitespace after ','
./luma/oled/device.py:382:16: E231 missing whitespace after ','
./luma/oled/device.py:382:18: E231 missing whitespace after ','
./luma/oled/device.py:383:35: E203 whitespace before ':'
```

**Comment by leragequit** [comment]
@thijstriemstra thank you, I understand and will fix those

**Comment by leragequit** [comment]
i'll still try to integrate the test cases once i get the time, but how woulld one restart the travis-ci client. i suspect that the failed test was due to an outage they had yesterday.

**Comment by rm-hull** [comment]
@leragequit Could you add your name to the list of contributors in https://github.com/rm-hull/luma.oled/blob/master/CONTRIBUTING.rst also?

**Comment by thijstriemstra** [comment]
@rm-hull can you put out a new release?

**Comment by rm-hull** [comment]
@thijstriemstra: not sure about layout changes made in 3e09ec7 - find it a bit hard to digest when the comments are stacked above each line. Any objections if I change that back?

**Comment by rm-hull** [comment]
----
Also it looks like (at least for now) that `settings['displayoffset']` and `settings['startline']` are both invariant, and we could just use `0x00` for both. Indeed, the `settings['height']` and `settings['width']` could also be replaced with `height-1` and `width-1`.

```python
        # Set column address
        self.command(0x15, settings['displayoffset'], settings['width'])
        # Set row address
        self.command(0x75, settings['displayoffset'], settings['height'])
```

becomes:

```python
        self.command(0x15, 0x00, width - 1)   # Set column address
        self.command(0x75, 0x00, height - 1)  # Set row address
```

Actually, is `width=0x6F` a mistake in https://github.com/rm-hull/luma.oled/pull/192/files#diff-5cdd2e38808395886cb0470afe406f72R344 ? 0x6F is 112 decimal ... should that be 0x5F (95 decimal)

**Pull request merged**